rotkehlchen/globaldb/handler.py

```diff
@@ -298,35 +298,53 @@ def get_assets_mappings(identifiers: List[str]) -> Dict[str, dict]:
         search_result = []
         query, bindings = GlobalDBHandler()._prepare_search_assets_query(filter_query)  # noqa:E501
-        with GlobalDBHandler().conn.read_ctx() as cursor:
-            cursor.execute(query, bindings)
-            for entry in cursor:
-                search_result.append({
-                    'identifier': entry[0],
-                    'name': entry[1],
-                    'symbol': entry[2],
-                })
+        with db.conn.read_ctx() as cursor:
+            with GlobalDBHandler().conn.read_ctx() as global_db_cursor:
+                global_db_cursor.execute(query, bindings)
+                found_eth = False
+                for entry in global_db_cursor:
+                    if db.get_settings(cursor).treat_eth2_as_eth is True and entry[0] in (A_ETH.identifier, A_ETH2.identifier):  # noqa:E501
+                        if found_eth is True:
+                            continue
+                        found_eth = True
+
+                    search_result.append({
+                        'identifier': entry[0],
+                        'name': entry[1],
+                        'symbol': entry[2],
+                    })
         return search_result
 
     @staticmethod
     def search_assets_levenshtein(
             filter_query: 'AssetsFilterQuery',
+            db: 'DBHandler',
             substring_search: str,
             limit: Optional[int],
     ) -> List[Dict[str, str]]:
         """Returns a list of asset details that match the search keyword using the Levenshtein distance approach."""  # noqa: E501
         search_result = []
         levenshtein_distances = []
         query, bindings = GlobalDBHandler()._prepare_search_assets_query(filter_query)  # noqa:E501
-        with GlobalDBHandler().conn.read_ctx() as cursor:
-            cursor.execute(query, bindings)
-            for entry in cursor:
-                lev_dist_name = levenshtein(substring_search, entry[0])
-                lev_dist_symbol = levenshtein(substring_search, entry[1])
+        with GlobalDBHandler().conn.read_ctx() as globaldb_cursor, db.conn.read_ctx() as cursor:
+            globaldb_cursor.execute(query, bindings)
+            found_eth = False
+            for entry in globaldb_cursor:
+                if db.get_settings(cursor).treat_eth2_as_eth is True and entry[0] in (A_ETH.identifier, A_ETH2.identifier):  # noqa:E501
+                    if found_eth is True:
+                        continue
+                    found_eth = True
+
+                # if an asset does not have a name and symbol, skip.
+                if entry[1] is None and entry[2] is None:
+                    continue
+
+                lev_dist_name = levenshtein(substring_search, entry[1]) if entry[1] is not None else 0  # noqa: E501
+                lev_dist_symbol = levenshtein(substring_search, entry[2]) if entry[2] is not None else 0  # noqa: E501
```

Comment:
here in the case of ETH2/ETH I belive that if entry is ETH2 before ETH you are further from it?

Reply:
I don't get you here.